### DISCUSSION TITLE
Artifacts around widgets when fullscreen

### DISCUSSION THREAD

**Discussion by khuttun** [emilk/egui#239] [Q&A]
Hi! I'm just getting started with egui and it seems like a really nice and well thought out library. I'm running into an issue though. When running my application fullscreen, I get this weird artifact around the widget I'm creating with egui:

![image](https://user-images.githubusercontent.com/12572951/111840709-88544d00-8905-11eb-8fe4-088fde0c3a3d.png)

When running it in a window, there's no artifact and the widget is displayed as it should with the shadow:

![image](https://user-images.githubusercontent.com/12572951/111840816-ab7efc80-8905-11eb-95e1-629720f96107.png)

Any idea what could cause this? I'm doing my own integration with wgpu and winit running on macOS.

**Comment by emilk** [comment]
I have no idea - never seen this before. I'm pretty sure the problem is not in `egui` though :)

Maybe a division by zero in your shader (e.g. if you are trying to unmultiply the alpha) ?

**Reply by khuttun** [reply]
Thanks for the reply. Yeah, it's probably not egui, I just figured that maybe someone here would have seen something similar. The funny thing is that the artifact goes away and everything is displayed correctly if I draw in a window with the same inner size as the fullscreen resolution is. 🤷‍♂️ I don't think there's any /0 possibility in [the shader](https://github.com/khuttun/labyrinth/blob/master/src/2d.frag), but maybe it's an issue somewhere else in my code. The debugging continues...

**Comment by khuttun** [comment]
Ok, so I discovered that the artifact goes away when I disable VSync from Quartz Debug. This leads me to believe that this is actually some macOS issue, possibly related to [this](https://apple.stackexchange.com/q/338319/399563).

**Reply by khuttun** [reply]
And also the artifact appears only when using the built-in Intel graphics chip, not with the Radeon Pro also in the MBP I'm testing with, so it seems to be the same issue as the one linked above.
